urllib3/connectionpool.py

```diff
@@ -98,6 +98,23 @@ def close(self):
 # This is taken from http://hg.python.org/cpython/file/7aaba721ebc0/Lib/socket.py#l252
 _blocking_errnos = set([errno.EAGAIN, errno.EWOULDBLOCK])
 
+# CPython 2.7 supports an opt-in parameter for response buffering
+# in HTTPConnection.getresponse
+# Rather than checking on each request, we inspect the code object
+# (if available) at import and only pass the argument if it appears
+# in the list of variable names
+def _request_httplib_response_buffering():
```

Comment:
This should use the inspect module. The only fault here is that you need conditions to handle differences between Python 3.3 and beyond and earlier. See [this](https://github.com/PyCQA/flake8/blob/master/src/flake8/utils.py#L293..L306) for an idea of how this could be done.